src/tree/node_children.rs

```diff
@@ -562,44 +568,63 @@ mod inner {
         /// Access to the nodes array.
         #[inline(always)]
         pub fn nodes(&self) -> &[Arc<Node>] {
-            unsafe { mem::transmute(&self.nodes[..(self.len())]) }
+            let len = self.len();
+            // SAFETY: MaybeUninit<T> is layout compatible with T, and
+            // the nodes from 0..len are guaranteed to be initialized
+            unsafe { mem::transmute(&self.nodes[..len]) }
```

Comment:
I like the added safety note, but moving the length out into a separate variable seems superfluous to me.  Unless there was something wrong with the code that I'm not seeing, I'd rather leave the code itself as-is.  Same goes for the other places where you split it out.